teamA/crud-express/heeyeon/app.js

```diff
@@ -0,0 +1,178 @@
+require('dotenv').config();
+
+const http = require('http');
+
+const express = require('express');
+const cors = require('cors');
+const morgan = require('morgan');
+const { DataSource } = require('typeorm');
+
+const myDataSource = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPOPRM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE
+})
+
+myDataSource.initialize()
+    .then(() => {
+        console.log('Data Source has been initialized!')
+    });
+
+const app = express()
+
+app.use(express.json());
+app.use(morgan('tiny'));
+
+// health check
+app.get('/ping', cors(), function (req, res, next) {
+    res.status(200).json({ message : 'pong '})
+});
+
+// 4. 전체 게시글 조회하기
+app.get('/posts/lists', async (req, res, next) => {
+
+    await myDataSource.query(
+        `SELECT
+            p.user_id As userId,
+            u.profile_image As userProfileImage,
+            p.id AS postingId,
+            p.posting_img_url As postingImageUrl,
+            p.content As postingContent
+        FROM users u, posts p WHERE p.user_id=u.id;`
+    ,(err, rows) => {
+            res.status(200).json({ "data" : rows });
+    })
+});
+
+// 5. 유저의 게시글 조회하기
+app.get('/posts/:userId', async (req, res, next) => {
+    const userId = req.params.userId;
+
+    const user = await myDataSource.query(
+        `SELECT
+            id AS userId,
+            profile_image AS userProfileImage
+        FROM users WHERE id=${userId};`
+    );
+
+    const post = await myDataSource.query(
+        `SELECT
+            id AS postingId,
+            posting_img_url AS postingImageUrl,
+            content AS postingContent
+        FROM posts WHERE user_id=${userId};`
+    );
+
+    user[0]["postings"] = post;
+    res.status(201).json({ "data" : user })
+
+});
+
+// 2. 유저 회원가입 하기
+app.post('/users/signup', async (req, res, next) => {
+    const {id, name, email, profile_image, password} = req.body
+
+    await myDataSource.query(
+        `INSERT INTO users(
+            name,
+            email,
+            profile_image,
+            password
+        ) VALUES (?, ?, ?, ?);
+        `,
+        [name, email, profile_image, password]
+    );
+
+    res.status(201).json({ message : "userCreated" })
+
+});
+
+// 3. 게시글 등록하기
+app.post('/posts/post', async (req, res, next) => {

```

Comment:
다음 공통 세션에 해당하는 내용 중 하나입니다.

위코드에서는 RESTful API 기반의 API를 개발하고 있으며, 엔드포인트 URI에는 동사를 제외한 명사로 구성하도록 권장하고 있습니다.

```suggestion
app.post('/posts', async (req, res, next) => {
```